packages/client/modules/userDashboard/components/OrgBilling/BillingForm.tsx

```diff
@@ -20,56 +26,74 @@ const StyledForm = styled('form')({
   alignItems: 'space-between'
 })
 
-const PaymentWrapper = styled('div')({
-  height: 160
-})
-
 const UpgradeButton = styled(PrimaryButton)<{isDisabled: boolean}>(({isDisabled}) => ({
   background: isDisabled ? PALETTE.SLATE_200 : PALETTE.SKY_500,
   color: isDisabled ? PALETTE.SLATE_600 : PALETTE.WHITE,
   boxShadow: 'none',
   marginTop: 16,
   width: '100%',
   elevation: 0,
-  '&:hover': {
+  '&:hover, &:focus': {
     boxShadow: 'none',
     background: isDisabled ? PALETTE.SLATE_200 : PALETTE.SKY_600
   }
 }))
 
-export default function BillingForm() {
+const ConfettiWrapper = styled('div')({
+  position: 'fixed',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)'
+})
+
+type Props = {
+  orgId: string
+}
+
+const BillingForm = (props: Props) => {
+  const {orgId} = props
   const stripe = useStripe()
   const elements = useElements()
   const [isLoading, setIsLoading] = useState(false)
+  const [isPaymentSuccessful, setIsPaymentSuccessful] = useState(false)
+  const atmosphere = useAtmosphere()
+  const {onError} = useMutationProps()
 
-  // TODO: implement in: https://github.com/ParabolInc/parabol/issues/7693
-  // look at: https://stripe.com/docs/payments/quickstart
-  const handleSubmit = async () => {
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    if (!stripe || !elements) return
+    setIsLoading(true)
+    const {setupIntent, error} = await stripe.confirmSetup({
+      elements,
+      redirect: 'if_required'
+    })
     setIsLoading(false)
+    if (error) return
+    const {payment_method: paymentMethodId, status} = setupIntent
+    if (status === 'succeeded' && typeof paymentMethodId === 'string') {
```

Comment:
-1 What about if not? If I test with a [declining card](https://stripe.com/docs/testing#declined-payments) 4000000000000002 or with a [3D Secure card](https://stripe.com/docs/testing#authentication-and-setup) 4000002760003184 and decline, then there is no message at all

Reply:
Thanks, updated!

By the way, 3D Secure cards won't create subscriptions in this PR. It'll be handled in https://github.com/ParabolInc/parabol/issues/7864